### PR TITLE
Add missing dependencies

### PR DESCRIPTION
**Pull request by samhedin**
For https://github.com/brotzeit/rustic/issues/150
`missing-imports` is just a list at the moment. What is the preferred way to prompt the user to select one of them? An alternative would be to just add all missing dependencies. I'm not sure how often a user would input several missing dependencies and then only want to add a subset of them. And in those cases, you could just not run `rustic-cargo-add-missing-import`.

### CODE REVIEW

**Comment by brotzeit** [comment]
I would say we should add all missing dependencies by default. When the user runs the command with prefix argument it should offer the user dependencies that he can choose manually.

**Comment by samhedin** [comment]
Good idea, I added it now.

File: rustic-cargo.el [hunk 1]
```diff
@@ -446,6 +446,22 @@ If running with prefix command `C-u', read whole command from minibuffer."
...
+Adds all missing imports by default.
+Use with 'prefix-arg` to select imports to add."
+  (interactive)
+  (let ((missing-imports (delete-dups
+                          (seq-reduce (lambda (missing-crates errortable)
+                                        (if (string= "E0432" (gethash "code" errortable))
+                                            (cons (nth 3 (split-string (gethash "message" errortable) "`")) missing-crates)
+                                          missing-crates))
+                                      (gethash (buffer-file-name) (lsp-diagnostics t)) '()))))
+    (rustic-run-cargo-command (format  "cargo add %s"
```
Comment (brotzeit) [inline rustic-cargo.el:460]: Why not use `rustic-cargo-add` ?

Comment (samhedin) [reply rustic-cargo.el:460]: `rustic-cargo-add` reads from the minibuffer, so I figured it was not meant for the case that's needed here. Is there a way to use `rustic-cargo-add` that I am missing? Of course, `rustic-cargo-add` could be modified to work with `rustic-cargo-add-missing-import`, but I think the end result would be worse than the current situation.

Comment (brotzeit) [reply rustic-cargo.el:460]: You're right, I haven't thought about it.

I think we should also handle the case when `missing-imports` is `nil`.

Comment (samhedin) [reply rustic-cargo.el]: Done!

**Comment by brotzeit** [comment]
Very handy. If you want you could also use `if-let`.

**Review by brotzeit — APPROVED** [review]
You can merge when you're happy with it.

**Comment by brotzeit** [comment]
Ah sorry, please add a note to the readme. And please mention that this function currently only works with lsp-mode.

**Comment by samhedin** [comment]
Good catch!

**Comment by samhedin** [comment]
> Very handy. If you want you could also use `if-let`.

Oh, I missed this comment. You are right, this would have been nicer. I am looking into eglot support, will fix there.

**Pull request merged**